src/static/helpers/customApi.test.ts

```diff
@@ -0,0 +1,262 @@
+
+    const runFetchPassedArgs = runFetchSpy.getCall(0).args;
+
+    const expectedPathParms = {
+      ...clientConfig.parameters,
+      ...options.customApiPathParameters,
+    };
+
+    expect(runFetchSpy.callCount).to.equal(1);
+    expect(runFetchPassedArgs[0]).to.equal("post");
+    expect(runFetchPassedArgs[1].client).to.deep.equal({
+      clientConfig: {
+        ...clientConfig,
+        baseUri: CUSTOM_API_DEFAULT_BASE_URI,
+        parameters: expectedPathParms,
+      },
+    });
+    expect(runFetchPassedArgs[1].pathParameters).to.deep.equal(
+      expectedPathParms
+    );
+    expect(runFetchPassedArgs[1].queryParameters).to.deep.equal({
+      ...options.parameters,
+      siteId: clientConfig.parameters?.siteId,
+    });
+    expect(runFetchPassedArgs[1].headers).to.deep.equal(options.headers);
+    expect(runFetchPassedArgs[1].rawResponse).to.equal(true);
+    expect(runFetchPassedArgs[1].body).to.equal(options.body);
+  });
+
+  it("uses path params from options and clientConfig, prioritizing options", async () => {
+    const copyClientConfig = {
+      ...clientConfig,
+      // Only shortCode will be used
+      parameters: {
+        endpointPath: "clientConfig_endpoint_path",
+        apiName: "clientConfig_api_name",
+        shortCode: "clientconfig_shortcode",
+        apiVersion: "v2",
+        organizationId: "clientConfig_organizationId",
+        siteId: "site_id",
+      },
+    };
+
+    const copyOptions = {
+      ...options,
+      // these parameters will be prioritzed
+      customApiPathParameters: {
+        endpointPath: "customApiPathParameters_endpoint_path",
+        apiName: "customApiPathParameters_api_name",
+        apiVersion: "v3",
+        organizationId: "customApiPathParameters_organizationId",
+      },
+    };
+
+    // nock interception should be using custom API path parameters from options
+    const { apiName, endpointPath, organizationId, apiVersion } =
+      copyOptions.customApiPathParameters;
+    // except shortcode since we didn't implement it in copyOptions.customApiPathParameters
+    const { shortCode } = copyClientConfig.parameters;
+
+    const nockBasePath = `https://${shortCode}.api.commercecloud.salesforce.com`;
+    const nockEndpointPath = `/custom/${apiName}/${apiVersion}/organizations/${organizationId}/${endpointPath}`;
+    nock(nockBasePath).post(nockEndpointPath).query(true).reply(200);
+
+    await callCustomEndpoint({
+      options: copyOptions,
+      clientConfig: copyClientConfig,
+    });
+
+    const runFetchPassedArgs = runFetchSpy.getCall(0).args;
+
+    const expectedPathParams = {
+      ...copyClientConfig.parameters,
+      ...copyOptions.customApiPathParameters,
+    };
+
+    expect(runFetchSpy.callCount).to.equal(1);
+    expect(runFetchPassedArgs[1].pathParameters).to.deep.equal(
+      expectedPathParams
+    );
+    expect(runFetchPassedArgs[1].client.clientConfig.parameters).to.deep.equal(
+      expectedPathParams
+    );
```

Comment:
Interesting thing, and I'm making an assumption, is that the function itself accomplishes this "prioritization" by using the spread operator, but we are also using the spread operator to create this "expectedPathParams". So it almost nullifies the test in a way. Maybe we just test directly the values one by one to see if they are the expected values? 

Let me know if I'm wrong in my assumption.